dashboard/app/helpers/milestone_helper.rb

```diff
@@ -0,0 +1,83 @@
+module MilestoneHelper
+  include ScriptLevelsHelper
+  include LevelsHelper
+
+  def milestone_response(options)
+    response = {
+      timestamp: DateTime.now.to_milliseconds
+    }
+    script_level = options[:script_level]
+    level = options[:level]
+
+    if script_level
+      response[:script_id] = script_level.script.id
+      response[:level_id] = level.id
+
+      previous_level = script_level.previous_level
+      if previous_level
+        response[:previous_level] = build_script_level_path(previous_level)
+      end
+
+      # if they solved it, figure out next level
+      if options[:solved?]
+        response[:total_lines] = options[:total_lines]
+        response[:new_level_completed] = options[:new_level_completed]
+        response[:level_path] = build_script_level_path(script_level)
+        script_level_solved_response(response, script_level)
+      else # not solved
+        response[:message] = 'try again'
+      end
+    else
+      response[:message] = 'no script provided'
+    end
+
+    if options[:level_source].try(:id)
+      response[:level_source] = level_source_url(id: options[:level_source].id)
+      response[:phone_share_url] = send_to_phone_url
```

Comment:
As we're cleaning up milestone posts it would make sense to move this to app init, `appOptions` or somewhere that doesn't send the same value down each time there is a milestone post!